src/cache/inmemory/entityStore.ts

```diff
@@ -211,7 +211,10 @@ export abstract class EntityStore implements NormalizedCache {
     });
     if (newData) {
       Object.keys(newData).forEach(dataId => {
-        this.merge(dataId, newData[dataId]);
+        const newDatum = newData[dataId];
+        if (newDatum !== void 0) {
+          this.merge(dataId, newDatum);
+        }
```

Comment:
Rather than changing/adding code to make the type system happy, I would recommend changing the types to reflect the behavior of the code. Of course, if you notice a bug in the code, feel free to make a quick change to fix it, especially if it helps with type checking!

In this particular case, there's no risk that `newDatum` will be undefined, so you can let the type system know that with a type cast expression:
```suggestion
      Object.keys(newData).forEach(dataId => {
        this.merge(dataId, newData[dataId] as StoreObject);
      });
```